pkgs/misc/uboot/default.nix

```diff
@@ -147,19 +147,26 @@ in {
     extraMeta.platforms = lib.platforms.linux;
 
     crossTools = true;
-    extraMakeFlags = [ "HOST_TOOLS_ALL=y" "NO_SDL=1" "cross_tools" ];
+    extraMakeFlags = [ "HOST_TOOLS_ALL=y" "NO_SDL=1" "cross_tools" "envtools" ];
 
     outputs = [ "out" "man" ];
 
     postInstall = ''
       installManPage doc/*.1
+
+      # from u-boot's tools/env/README:
+      # "You should then create a symlink from fw_setenv to fw_printenv. They
+      # use the same program and its function depends on its basename."
+      ln -s $out/bin/fw_printenv $out/bin/fw_setenv
     '';
+
     filesToInstall = [
       "tools/dumpimage"
       "tools/fdtgrep"
       "tools/kwboot"
       "tools/mkenvimage"
       "tools/mkimage"
+      "tools/env/fw_printenv"
```

Comment:
Why is only one of the new tools added here?

Reply:
The comment in the postInstall step explains why. The programs are the same, just a different argv[0].

Reply:
I see that, but I thought filesToInstall applies after the package is built. Meaning the symlink is not copied.